File: problemas/iniciante/1099/1099.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.lang.Math;

public class Main {
    public static int numeroImpares(int X, int Y){
        return (Y - X) / 2 + 1;
    }

    public static int somaPA(int a0, int an){
        return (a0 + an) * numeroImpares(a0, an) / 2;
    }

    public static void main(String[] args) throws IOException {
        InputStreamReader ir = new InputStreamReader(System.in);
        BufferedReader in = new BufferedReader(ir);
    
        int N = Integer.parseInt(in.readLine());
        for(int i = 0; i < N; ++i){
            String[] entrada = in.readLine().trim().split(" ");
            int X = Integer.parseInt(entrada[0]);
            int Y = Integer.parseInt(entrada[1]);

            if(X > Y){
                int temp = X;
                X = Y;
                Y = temp;
            }

            X += (Math.abs(X) % 2) + 1;
            Y -= (Math.abs(Y) % 2) + 1;

            System.out.printf("%d\n", X > Y ? 0 : somaPA(X, Y));
        }
    }
}

File: problemas/iniciante/1099/1099.scala
object Main {
  def numeroImpares(x: Int, y: Int): Int = {
    return (y - x)/2 + 1
  }

  def somaPA(a1: Int, an: Int, n: Int): Int = {
    return ((a1 + an) * n)/2
  }

  def main(args: Array[String]): Unit = {
    val n = io.StdIn.readLine().toInt

    for(i <- 1 to n){
      var entrada = io.StdIn.readLine().split(" ").map(_.toInt)
      var x = entrada(0)
      var y = entrada(1)
    
      if(x > y){
        val temp = x
        x = y
        y = temp
      }
    
      x += (x.abs % 2) + 1
      y -= (y.abs % 2) + 1

      if(x > y){
        printf("0\n")
      }else{
        printf("%d\n", somaPA(x, y, numeroImpares(x, y)))
      }
    }
  }
}
